### PR TITLE
LINK-99, LINK-100, LINK-178 | Show all event form fields at same page

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Validating draft events and scrolling to first error when clicking "Save draft" button
Validating published event snd scrolling to first error when clicking "Publish event" button
Showing all event related fields at same page
Removed deprecated components

## Closes 
[LINK-99](https://helsinkisolutionoffice.atlassian.net/browse/LINK-99)
[LINK-100](https://helsinkisolutionoffice.atlassian.net/browse/LINK-100)
[LINK-178](https://helsinkisolutionoffice.atlassian.net/browse/LINK-178)

## Screenshots
Screenshot 2021-01-11 at 12 38 18

## Testing
You can test validation and scrolling to first error on http://localhost:3000/en/event/create. Needs log in. After successful validation you can see "Sign in to perform this action" toast message. This is expected because at the moment authentication is working only locally with api keys which is commented out.

### CODE REVIEW

File: src/domain/event/CreateEventPage.tsx [hunk 1]
```diff
@@ -128,49 +157,104 @@ const CreateEventPage: React.FC = () => {
...
+      validateOnChange={true}
     >
-      {({ values: { type, ...restValues }, setErrors }) => {
+      {({ values: { type, ...restValues }, setErrors, setTouched }) => {
+        const getFocusableFieldId = (fieldName: string): string => {
+          // For the select elements, focus the toggle button
+          if (
+            fieldName === EVENT_FIELDS.KEYWORDS ||
+            fieldName === EVENT_FIELDS.LOCATION ||
+            fieldName === EVENT_FIELDS.SUPER_EVENT
```
Comment (Sartonon) [inline src/domain/event/CreateEventPage.tsx:181]: maybe create select elements array and use .find function here

File: src/domain/event/buttonPanel/ButtonPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      <Container>
+        <FormContainer>
+          <div className={styles.buttonsRow}>
+            <div className={styles.draftButtonColumn}>
+              <Button
+                disabled={!authenticated}
+                fullWidth={true}
+                onClick={onSaveDraft}
+                title={
+                  !authenticated ? t('authentication.noRightsCreateEvent') : ''
```
Comment (Sartonon) [inline src/domain/event/buttonPanel/ButtonPanel.tsx:33]: I would prefer to avoid negation for these but maybe it makes sense here?

**Review by Sartonon — COMMENTED** [review]
Looks very good. Couple suggestions/notes :)

File: src/common/components/formLanguageSelector/FormLanguageSelector.tsx [hunk 3]
```diff
@@ -6,9 +6,40 @@ import React from 'react';
...
+
+const Item: React.FC<ItemProps> = ({
+  isCompleted,
+  isSelected,
+  language,
+  onClick,
+}) => {
+  const handleClick = () => {
+    onClick(language.value);
+  };
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/formLanguageSelector/FormLanguageSelector.tsx]: Changed. But I agree that there is no right way.

File: src/common/components/keywordSelector/KeywordSelector.tsx [hunk 4]
```diff
@@ -52,6 +52,8 @@ const KeywordSelector: React.FC<KeywordSelectorProps> = ({
   value,
   ...rest
 }) => {
+  let timer: NodeJS.Timeout;
+  const isMounted = React.useRef(false);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/keywordSelector/KeywordSelector.tsx]: Very nice idea. Added an useIsMounted hook and used it whenever possible.

File: src/common/components/keywordSelector/KeywordSelector.tsx [hunk 5]
```diff
@@ -69,8 +71,10 @@ const KeywordSelector: React.FC<KeywordSelectorProps> = ({
   });
 
   const handleFilter = (items: OptionType[], inputValue: string) => {
-    setTimeout(() => {
-      setSearch(inputValue);
+    timer = setTimeout(() => {
+      if (isMounted.current) {
+        setSearch(inputValue);
+      }
     }, 0);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/keywordSelector/KeywordSelector.tsx]: Yeah, I think that 0 is the default value. Removed redundant 0 and fixed TS type

File: src/common/components/keywordSelector/KeywordSelector.tsx [hunk 6]
```diff
@@ -110,6 +114,16 @@ const KeywordSelector: React.FC<KeywordSelectorProps> = ({
...
 
+  React.useEffect(() => {
+    isMounted.current = true;
+
+    return () => {
+      clearTimeout(timer);
+      isMounted.current = false;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/keywordSelector/KeywordSelector.tsx]: Changed as suggested

File: src/common/components/placeSelector/PlaceSelector.tsx [hunk 7]
```diff
@@ -59,6 +59,8 @@ const PlaceSelector: React.FC<PlaceSelectorProps> = ({
   value,
   ...rest
 }) => {
+  let timer: NodeJS.Timeout;
+  const isMounted = React.useRef(false);
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/common/components/placeSelector/PlaceSelector.tsx]: Used new useIsMounted hook here as well

File: src/domain/event/CreateEventPage.tsx (hunk 1 above)
Comment (jorilindell) [reply src/domain/event/CreateEventPage.tsx]: Nice idea. Added an array and used it here to find is the fieldName in the list or not

File: src/domain/event/buttonPanel/ButtonPanel.tsx (hunk 2 above)
Comment (jorilindell) [reply src/domain/event/buttonPanel/ButtonPanel.tsx]: Changed to `authenticated ? '' : t('authentication.noRightsCreateEvent')`. I did similar change to publish button as well

**Review by Sartonon — APPROVED** [review]
Great!

**Pull request merged**